### PR TITLE
Change imagenet data script to include training data

### PR DESCRIPTION
**Pull request by MaxVanDijck**
Added script in readme to manage extraction of both training and validation data in reference to: https://github.com/pytorch/examples/issues/933

### CODE REVIEW

File: imagenet/README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ This implements training of popular model architectures, such as ResNet, AlexNet
 - Install PyTorch ([pytorch.org](http://pytorch.org))
 - `pip install -r requirements.txt`
 - Download the ImageNet dataset from http://www.image-net.org/
-    - Then, and move validation images to labeled subfolders, using [the following shell script](https://raw.githubusercontent.com/soumith/imagenetloader.torch/master/valprep.sh)
+    - Then, move and extract the training and validation images to labeled subfolders, using [the following shell script](https://gist.github.com/MaxVanDijck/39531343d1c2ff6a0ee3304d8dee917f)
```
Comment (msaroufim) [inline imagenet/README.md:10]: Would suggest adding your script to source control since it's pretty readable and could you add some brief explanation or proof as to how your script doesnt change the intended behavior in soumith's old script?

**Review by msaroufim — CHANGES_REQUESTED** [review]

File: imagenet/README.md (hunk 1 above)
Comment (MaxVanDijck) [reply imagenet/README.md:10]: previously you have to extract the validation .tar file inside a directory named 'val' and then run Soumith's script

Soumith script:
(creates directories for categories)
`mkdir -p n01440764`
(moves images into directories)
`mv ILSVRC2012_val_00000293.JPEG n01440764/`

this results in 
val/
├── n01440764
│   ├── ILSVRC2012_val_00000293.JPEG
│   ├── ......
├── ......



New script uses extraction method from archive here modified to extract images to 'imagenet/train' and 'imagenet/val'
https://github.com/facebookarchive/fb.resnet.torch/blob/master/INSTALL.md#download-the-imagenet-dataset

for training data:
create directory, move .tar file and change directory to imagenet/train
`mkdir imagenet/train && mv ILSVRC2012_img_train.tar imagenet/train/ && cd imagenet/train`
extract training set into dir and remove compressed file
`tar -xvf ILSVRC2012_img_train.tar && rm -f ILSVRC2012_img_train.tar`

this results in a directory of compressed .tar files, one for each folder of images per each category

find all .tar files, create directory with same name of tar file, decompress and copy contents of tar file to new directory, remove .tar file
`find . -name "*.tar" | while read NAME ; do mkdir -p "${NAME%.tar}"; tar -xvf "${NAME}" -C "${NAME%.tar}"; rm -f "${NAME}"; done`

this results in a directory full of new directories, each with the same name of the .tar file they were created from and containing the contents of the .tar file in this case images
eg.
train/
├── n01440764
│   ├── n01440764_10026.JPEG
│   ├── n01440764_10027.JPEG
│   ├── ......
├── ......

for the validation data:
create directory, move .tar file, change directory to imagenet/val and extract validation data
`mkdir imagenet/val && mv ILSVRC2012_img_val.tar imagenet/val/ && cd imagenet/val && tar -xvf ILSVRC2012_img_val.tar`
then simply wget and run Soumith's script for validation data
`wget -qO- https://raw.githubusercontent.com/soumith/imagenetloader.torch/master/valprep.sh | bash`

resulting in 
val/
├── n01440764
│   ├── ILSVRC2012_val_00000293.JPEG
│   ├── ......
├── ......


**TLDR** In essence, validation set is managed in the same way except you don't have to extract before running Soumith's script
train set is managed using the names of the .tar files for each category after the initial ILSVRC2012_img_train.tar is extracted

Comment (msaroufim) [reply imagenet/README.md]: Nice do you mind summarizing this thread in the README for this example or in your shell script?

**Review by msaroufim — CHANGES_REQUESTED** [review]
Let's just integrate the notes from your comment into either the README or the shell script to explain this change

**Review by msaroufim — APPROVED** [review]
